### ISSUE TITLE
Consider adding 'views_infinite_scroll’

### ISSUE THREAD

**Issue by isalmanhaider** [phenaproxima/starshot-prototype#92]
Integrating the Views Infinite Scroll module will significantly improve user experience on views that load a lot of content, enhancing the overall usability of Drupal. This feature is beneficial for both beginner and professional developers exploring Drupal and Views in Drupal, offering a modern and user-friendly way to navigate through content-dense pages. Since in addition to pagination, it allows content to load on the fly as you scroll via AJAX.

https://www.drupal.org/project/views_infinite_scroll

**Comment by phenaproxima** [comment]
No specific objection to adding Views Infinite Scroll, but let's add it when we have an actual use case for it -- that is, when we have a specific view that would benefit from it. I don't think all views are the best fit for infinite scrolling (administrative views are probably just fine with regular pagination), but when/if we have one that whose usability would be enhanced by an infinite scroll, I'm all for it.

Leaving this open just in case anyone has thoughts about any views _already_ in Starshot which might benefit.

**Comment by phenaproxima** [comment]
I think infinite scrolling might make sense for something like the media library, where pagination is probably more annoying than useful. What do you think? Maybe we could have a recipe here which sets up infinite scrolling in the `media_library` view.

**Comment by pameeela** [comment]
I think we should probably postpone discussions around adding new contrib. There will be a process for reviewing modules to be included that is yet to be defined.

I think this module is definitely in consideration, but part of that will be weighing up the benefit vs cost of having another module. If we are only adding it for the media library, that seems like somewhat of a weak case but I could see it being useful for other views as well.

**Comment by gitressa** [comment]
I agree with postponing adding further contrib modules for now @pameeela.

As I commented elsewhere, best practice in Drupal is to not install a lot of modules, since it adds to future maintenance. Like Ted Bowman once said, "there's a module for that, don't use it". Starshot should be a robust starting point, showcasing and exemplifying best practices, delivering a Drupal web site ready for production "Out of The Box", like WordPress and SquareSpace do. A goal of covering 80% of popular and often used web site features could be used as a metric?

Here are the modules already added, and some of the modules under consideration, or rejected:

## Included

- https://www.drupal.org/project/address
- https://www.drupal.org/project/antibot
- https://www.drupal.org/project/automatic_updates
- https://www.drupal.org/project/coffee
- https://www.drupal.org/project/crop
- https://www.drupal.org/project/ctools
- https://www.drupal.org/project/default_content
- https://www.drupal.org/project/diff
- https://www.drupal.org/project/easy_breadcrumb
- https://www.drupal.org/project/eca
- https://www.drupal.org/project/focal_point
- https://www.drupal.org/project/geocoder
- https://www.drupal.org/project/geofield
- https://www.drupal.org/project/gin_toolbar
- https://www.drupal.org/project/honeypot
- https://www.drupal.org/project/leaflet
- https://www.drupal.org/project/leaflet_more_maps
- https://www.drupal.org/project/linkit
- https://www.drupal.org/project/media_entity_download
- https://www.drupal.org/project/media_file_delete
- https://www.drupal.org/project/metatag
- https://www.drupal.org/project/pathauto
- https://www.drupal.org/project/project_browser
- https://www.drupal.org/project/quick_node_clone
- https://www.drupal.org/project/redirect
- https://www.drupal.org/project/sam
- https://www.drupal.org/project/scheduler
- https://www.drupal.org/project/simple_sitemap
- https://www.drupal.org/project/smart_date
- https://www.drupal.org/project/token
- https://www.drupal.org/project/type_tray
- https://www.drupal.org/project/uli_custom_workflow
- https://www.drupal.org/project/webform

## Under consideration

- #17
- #72
- #82
- #83 
- #92 

## Rejected

- #21
- #89
- #86 

Two other module discussion issues:

- #2
- #11
